cli/azd/internal/repository/infra_confirm.go

```diff
@@ -370,6 +370,7 @@ func (i *Initializer) buildInfraSpecByAzureDep(
 			EventHubNames:             dependency.Names,
 			AuthUsingConnectionString: authType == scaffold.AuthType_PASSWORD,
 			AuthUsingManagedIdentity:  authType == scaffold.AuthType_TOKEN_CREDENTIAL,
+			UseKafka:                  dependency.UseKafka,
```

Comment:
Seem when `UseKafka = tue`, it's not necessary to `chooseAuthTypeByPrompt`, is that right? If that's right, please update related code.

Reply:
How about deleting `AuthUsingConnectionString`, `AuthUsingManagedIdentity` and `UseKafka` in `AzureDepEventHubs`, and use `connectionType = connectionString / managedIdentity / kafkaBroker` instead?

Reply:
I don't get your words. When using Kafka, it also needs to choose MI auth or connection string auth. So I think it's necessary, right?

Reply:
Or, we can warp the `spring.cloud.stream.kafka.binder.brokers` config when using MI, because it needs to know the endpoint when using MI. The connection string has already contained endpoint info, so it doesn't need.